react/components/BackToTopButton/index.tsx

```diff
@@ -0,0 +1,74 @@
+import React, { useCallback, useState, useEffect } from 'react'
+import { Button } from 'vtex.styleguide'
+import { FormattedMessage } from 'react-intl'
+import styles from './styles.css'
+import { Icon } from 'vtex.store-icons'
+
+interface Props {
+  showAfterPixelY?: number
+  rightPosition?: string
+  bottomPosition?: string
+  isIconStyle?: boolean
+}
+
+const BackToTopButton: StorefrontFC<Props> = props => {
+  const handleBackTop = useCallback(() => {
+    window.scroll({ top: 0, left: 0, behavior: 'smooth' })
+  }, [])
+  const [isShowed, setIsShowed] = useState(false)
+  const [scrollY, setScrollY] = useState(0)
+  useEffect(() => {
+    function watchScroll() {
+      window.addEventListener('scroll', scrollValue)
+    }
+    watchScroll()
+    return () => {
+      window.removeEventListener('scroll', scrollValue)
+    }
+  })
```

Comment:
Why you're doing it in every render? why not just

```suggestion
  useEffect(() => {
    window.addEventListener('scroll', scrollValue)
    return () => {
      window.removeEventListener('scroll', scrollValue)
    }
  }, [])
```

Reply:
Could you address this change too?